syntax = "proto3";

package gladys.types;

// ============================================================================
// SHARED TYPES - Cross-service boundary types
// ============================================================================
//
// PRINCIPLE: If a type is CREATED by service A and CONSUMED by service B,
// it belongs here. This prevents class mismatch errors when copying between
// proto messages from different packages.

// Salience dimensions for event evaluation.
// Used in both Event (orchestrator routing) and EpisodicEvent (memory storage).
message SalienceVector {
    float threat = 1;               // 0-1
    float opportunity = 2;          // 0-1
    float humor = 3;                // 0-1
    float novelty = 4;              // 0-1
    float goal_relevance = 5;       // 0-1
    float social = 6;               // 0-1
    float emotional = 7;            // -1 to 1
    float actionability = 8;        // 0-1
    float habituation = 9;          // 0-1
}

// ============================================================================
// HEALTH CHECK - Shared health messages for all services
// ============================================================================

enum HealthStatus {
    HEALTH_STATUS_UNKNOWN = 0;
    HEALTH_STATUS_HEALTHY = 1;
    HEALTH_STATUS_UNHEALTHY = 2;
    HEALTH_STATUS_DEGRADED = 3;     // Partially functional
}

message GetHealthRequest {}

message GetHealthResponse {
    HealthStatus status = 1;
    string message = 2;             // Optional: reason for status
}

message GetHealthDetailsRequest {}

message GetHealthDetailsResponse {
    HealthStatus status = 1;
    int64 uptime_seconds = 2;
    map<string, string> details = 3;  // Service-specific key-value pairs
}
